server/src/main/java/org/apache/druid/server/coordinator/loading/SegmentLoadingConfig.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.server.coordinator.loading;
+
+import org.apache.druid.java.util.common.logger.Logger;
+import org.apache.druid.server.coordinator.CoordinatorDynamicConfig;
+
+/**
+ * Contains recomputed configs from {@link CoordinatorDynamicConfig} based on
+ * whether {@code smartSegmentLoading} is enabled or not.
+ */
+public class SegmentLoadingConfig
+{
+  private static final Logger log = new Logger(SegmentLoadingConfig.class);
+
+  private final int maxSegmentsInLoadQueue;
+  private final int replicationThrottleLimit;
+  private final int maxReplicaAssignmentsInRun;
+  private final int maxLifetimeInLoadQueue;
+
+  private final int maxSegmentsToMove;
+  private final int percentDecommSegmentsToMove;
+
+  private final boolean useRoundRobinSegmentAssignment;
+  private final boolean emitBalancingStats;
+
+  public SegmentLoadingConfig(CoordinatorDynamicConfig dynamicConfig, int numUsedSegments)
```

Comment:
design nit: a constructor should generally define the "dependency relationship" of a class.  I.e. the things passed in on the constructor are the things that the current class is dependent upon.  This constructor is doing a bunch of work, that work is dependent on the `CoordinatorDynamicConfig` object, but `SegmentLoadingConfig` is not dependent on the object.

In this case, it would be preferable for this to be a static method and the constructor just take all of the various values.